clients/admin-ui/src/pages/consent/privacy-notices/index.tsx

```diff
@@ -1,43 +1,215 @@
-    <Box mb={4}>
-      <Heading fontSize="2xl" fontWeight="semibold" mb={2} data-testid="header">
-        Manage privacy notices
-      </Heading>
-      <Box>
-        <Breadcrumb
-          fontWeight="medium"
-          fontSize="sm"
-          color="gray.600"
-          data-testid="breadcrumbs"
+import {
+  PRIVACY_NOTICES_ROUTE,
+  SYSTEM_ROUTE,
+} from "~/features/common/nav/v2/routes";
+import {
+  DateCell,
+  FidesTable,
+  MechanismCell,
+  MultiTagCell,
+  TitleCell,
+  WrappedCell,
+} from "~/features/common/table";
+import {
+  selectAllPrivacyNotices,
+  selectPage,
+  selectPageSize,
+  useGetAllPrivacyNoticesQuery,
+  usePatchPrivacyNoticesMutation,
+} from "~/features/privacy-notices/privacy-notices.slice";
+import { PrivacyNoticeResponse, ScopeRegistryEnum } from "~/types/api";
+
+type EnableCellProps<T extends object> = CellProps<T, boolean> & {
+  onToggle: (data: any) => Promise<any>;
+};
+
+export const EnableCell = <T extends object>({
+  value,
+  column,
+  row,
+  onToggle,
+}: EnableCellProps<T>) => {
+  const modal = useDisclosure();
+  const handlePatch = async ({ enable }: { enable: boolean }) => {
+    // @ts-ignore
+    await onToggle([{ id: row.original.id, disabled: !enable }]);
+  };
+
+  const handleToggle = async (event: ChangeEvent<HTMLInputElement>) => {
+    const { checked } = event.target;
+    if (checked) {
+      await handlePatch({ enable: true });
+    } else {
+      modal.onOpen();
+    }
+  };
+
+  return (
+    <>
+      <Switch
+        colorScheme="complimentary"
+        isChecked={!value}
+        data-testid={`toggle-${column.Header}`}
+        /**
+         * It's difficult to use a custom column in react-table 7 since we'd have to modify
+         * the declaration file. However, that modifies the type globally, so our datamap table
+         * would also have issues. Ignoring the type for now, but should potentially revisit
+         * if we update to react-table 8
+         * https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/59837
+         */
+        // @ts-ignore
+        disabled={column.disabled}
+        onChange={handleToggle}
+      />
+      <ConfirmationModal
+        isOpen={modal.isOpen}
+        onClose={modal.onClose}
+        onConfirm={() => {
+          handlePatch({ enable: false });
+          modal.onClose();
+        }}
+        title="Disable privacy notice"
+        message={
+          <Text color="gray.500">
+            Are you sure you want to disable this privacy notice? Disabling this
+            notice means your users will no longer see this explanation about
+            your data uses which is necessary to ensure compliance.
+          </Text>
+        }
+        continueButtonText="Confirm"
+        isCentered
+        icon={<WarningIcon color="orange.100" />}
+      />
+    </>
+  );
+};
+
+const PrivacyNoticesPage = () => {
+  // Subscribe to get all privacy notices
+  const page = useAppSelector(selectPage);
```

Comment:
I have no basis for this, but it kind of bothers me when our `index.tsx` files have too much going on in them. I prefer them to just have the general layout, breadcrumbs, headings, maybe a loading/error state, and then leave the rest of their logic to a separate component. 

I find that it's easier to reason about a feature if all of its logic is together in the `features` folder, so that's how most of our other pages work (see `pages/systems/index.tsx`, `pages/datamap/index.tsx`, etc.).

in this case, I'd probably still have a `PrivacyNoticesTable.tsx` which has some of this "deep" logic with populating the table and then keep `index.tsx` just render `<PrivacyNoticesTable />` with minimal props.

this is probably just a personal preference though :)